### PR TITLE
[PTX-25079] Retry Device path validation

### PR DESCRIPTION
**Pull request by sayalasomayajula-px**
**What this PR does / why we need it**:

We need to retry the device path validation when px is upgrading , that is because right now volume device path is showing node name when node to which it is attached is down

**Which issue(s) this PR fixes** (optional)
Closes #
https://purestorage.atlassian.net/browse/PTX-25079

**Special notes for your reviewer**:
PASS LOGS: https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-pxe-byoc/3866/display/redirect

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -1403,10 +1421,8 @@ func (d *portworx) ValidateCreateVolume(volumeName string, params map[string]str
 	if vol.Spec.ProxySpec != nil && vol.Spec.ProxySpec.ProxyProtocol == api.ProxyProtocol_PROXY_PROTOCOL_PURE_BLOCK {
 		// Checking the device path when state is attached
 		if vol.State == api.VolumeState_VOLUME_STATE_ATTACHED && !strings.Contains(vol.DevicePath, DeviceMapper) {
-			return &ErrFailedToInspectVolume{
-				ID:    volumeName,
-				Cause: fmt.Sprintf("Failed to validate device path [%s]", vol.DevicePath),
-			}
+			_, err := task.DoRetryWithTimeout(volumeInspectDevicePath, 5*time.Minute, inspectVolumeRetryInterval)
+			log.FailOnError(err, "Failed to validate the device path for a volume")
```
Comment (lsrinivas-pure) [inline drivers/volume/portworx/portworx.go:1425]: return the error

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -1403,10 +1421,8 @@ func (d *portworx) ValidateCreateVolume(volumeName string, params map[string]str
 	if vol.Spec.ProxySpec != nil && vol.Spec.ProxySpec.ProxyProtocol == api.ProxyProtocol_PROXY_PROTOCOL_PURE_BLOCK {
 		// Checking the device path when state is attached
 		if vol.State == api.VolumeState_VOLUME_STATE_ATTACHED && !strings.Contains(vol.DevicePath, DeviceMapper) {
-			return &ErrFailedToInspectVolume{
-				ID:    volumeName,
-				Cause: fmt.Sprintf("Failed to validate device path [%s]", vol.DevicePath),
-			}
+			_, err := task.DoRetryWithTimeout(volumeInspectDevicePath, 5*time.Minute, inspectVolumeRetryInterval)
+			return fmt.Errorf("Failed to validate the device path for a volume [%s] : %v", volumeName, err)
```
Comment (vumesh-px) [inline drivers/volume/portworx/portworx.go:1425]: is there a reason to change the way we are returning? Can we retain `return &ErrFailedToInspectVolume`

**Review by vumesh-px — CHANGES_REQUESTED** [review]

File: drivers/volume/portworx/portworx.go (hunk 2 above)
Comment (sayalasomayajula-px) [reply drivers/volume/portworx/portworx.go]: yes sure, we can do that ,i have reverted that

**Review by vumesh-px — APPROVED** [review]
LGTM

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Pull request merged**